downstream.go

```diff
@@ -891,45 +891,75 @@ func (dc *downstreamConn) handleMessageRegistered(msg *irc.Message) error {
 				Command: irc.ERR_USERSDONTMATCH,
 				Params:  []string{dc.nick, "Cannot change mode for other users"},
 			}}
-		} else if modeStr != "" {
-			params := []string{upstreamName, modeStr}
-			params = append(params, msg.Params[2:]...)
-			uc.SendMessage(&irc.Message{
-				Command: "MODE",
-				Params:  params,
-			})
 		} else {
-			ch, ok := uc.channels[upstreamName]
-			if !ok {
-				return ircError{&irc.Message{
-					Command: irc.ERR_NOSUCHCHANNEL,
-					Params:  []string{dc.nick, name, "No such channel"},
-				}}
-			}
+			if modeStr != "" {
```

Comment:
This change is unrelated to this commit. I dropped it while applying the patch to master.